### PR TITLE
Chat Threads

### PR DESCRIPTION
**Pull request by AliceVlasov**
Define a class to handle multiple chat threads and implement functions necessary to only use the default thread.

Defined the Thread class to handle individual threads and also keep track of all the chatroom conversation threads. Defined empty functions that are anticipated to be necessary to implement this feature, both in Thread.js and chat.js. Created basic HTML elements that will be used for the thread menu where users can switch between threads and create new threads. 

Thread objects are in charge of storing their name and the DOM object containing all of the chat messages. as well as hiding and revealing themselves on the page, and building the message DOM elements. The Thread class keeps track of all existing threads of this chatroom, which thread is currently visible, setting default values and creating new Thread objects.

All chatrooms automatically build and display the default chat thread and since messages are not currently stored with a thread name, the default thread is used to display all messages.

### CODE REVIEW

File: src/main/webapp/Thread.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    title.innerHTML = this.name_;
+
+    let messages = makeEl("div", Thread.THREAD_MESSAGES);
+    this.messageWrapper_ = messages;
+
+    wrapper.appendChild(title);
+    wrapper.appendChild(messages);
+  }
+
+  /** Builds the menu item for this thread */
```
Comment (salehhamadeh) [inline src/main/webapp/Thread.js]: Please add a TODO or `@abstract` if it's meant for overriding

File: src/main/webapp/threadHandler.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+ * @param {String} threadName the name of the thread
+ */
+function addThread(threadName) {
+  let threadObj = new Thread(threadName);
+  allThreads[threadName] = threadObj;
+
+  chatWrapper.appendChild(threadObj.getThreadWrapper());
+}
+
+/** Makes the DOM element that allows users to create new elements */
```
Comment (salehhamadeh) [inline src/main/webapp/threadHandler.js]: Please either remove unused methods and add them when you use them, or add TODOs

Comment (AliceVlasov) [reply src/main/webapp/threadHandler.js]: The empty methods were for planning purposes so I'll turn them into TODOs

**Review by a-stepin — APPROVED** [review]

**Review by alexanderlopez — APPROVED** [review]

**Review by salehhamadeh — APPROVED** [review]

**Review by Felomeng — APPROVED** [review]

File: src/main/webapp/chat.js [hunk 3]
```diff
@@ -116,9 +126,10 @@ function addChatComment() {
       chatInput.value = "";
 
       if (connection) {
-          connection.send(JSON.stringify(commentObj));
-      }
+        connection.send(JSON.stringify(commentObj));
+      }
```
Comment (Felomeng) [inline src/main/webapp/chat.js]: nit: remove extra space.

**Review by Felomeng — APPROVED** [review]

**Pull request merged**